### ISSUE TITLE
HTML page breaks not rendered on print

### ISSUE THREAD

**Issue by scbarton** [jupyter/nbconvert#607]
In order to control pagination when generating a PDF, I have tried using the CSS directive `page-break-after:always` to force page breaks.  I then generate HTML using File > Print Preview in Jupyter and print from the browser.  

For example, as described [here](https://stackoverflow.com/questions/38884142/best-way-to-print-jupyter-notebook-slides-how-to-put-a-page-break), I might add a markdown cell containing
```
<p style="page-break-after:always;"></p>
```

However, these page break directives are not followed by any browser.  I have also tried the fix suggest in [ipython/ipython issue #5115](https://github.com/ipython/ipython/issues/5115) to no avail.

**Comment by mpacer** [comment]
If you use the print to PDF function directly does \pagebreak on a newline
with new lines above and below not work? I'll try this in a bit but pandoc
should be catching that LaTeX and then correctly converting it into the PDF.

**Comment by scbarton** [comment]
Yes, that does work, thank you!  It would be helpful to include that in the manual.  

I still think it would be useful to get it working in the HTML version, because of big differences in output styles between HTML and Latex.

**Comment by p2ya** [comment]
How was this fixed at the end ? I tried <p style="page-break-after:always;"></p> in markdown but it doesn't work

**Comment by mpacer** [comment]
The solution is to use LaTeX under the hood and put

```

\pagebreak

```

wherever you want a pagebreak in pdf output… Note: this doesn't work for html export at the same time, there's no good solution for that.

**Comment by p2ya** [comment]
I just need the page break in pdf.
At this stage I am converting notebook to html (jupyter nbconvert --to html noebook.ipynb) and then use https://pdfcrowd.com to get pdf.
I edited css file and added markdowns and it doesn't work.
Any suggestion for this method to make it work ?

**Comment by jowagner** [comment]
Related: issue #402

**Comment by jowagner** [comment]
It works if you move the line `<p><p style="page-break-after:always;"></p></p>` up above the line `<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">` in the intermediate html. (I should also mention that I removed the redundant `<p>` in `</p>`, in case this makes a difference.)

One of the surrounding `div` tags has `page-break-inside: avoid;` and this seems to be stronger. Briefly looking into CSS, there seems to be no way to balance different preferences, e.g. ability to change the weight of the vote of an element and a way to specify how an element should vote depending on whether the break would be before, inside or after the element.

**Comment by romanorac** [comment]
I wrote an Ebook in Jupyter Notebooks. Each chapter was in a separate Notebook. Then I converted a Notebook to HTML and then to PDF. The final step was to merge those PDFs. This was only was I was able to get page breaks in PDF.

I wrote a script that automates this. You can read more about it here: https://towardsdatascience.com/transform-jupyter-notebook-to-an-ebook-ef3a9d32ac4f.

**Comment by scbarton** [comment]
Here's one way to achieve page breaks in HTML (not latex) using `page-break-before`. This solution has some flaws, mentioned at the end, but it works. 

1. Add the  following to the custom.css file. Mine is in `/Users/scott/.jupyter/custom/custom.css`:
```
@media print {
  div.inner_cell {display: block}
}  
```
2. In the notebook, where page breaks are desired, insert the following in a markdown cell:
```
<span style="page-break-before: always !important">⚀</span>
```
I use the `⚀` symbol to indicate that a page break has been added. Feel free to use any text or none at all.

3. Create an HTML version of the notebook, either by using "Print Preview" or "Download as HTML" in the File menu, or by using `nbconvert`.

4. Open and print the created HTML file in *Firefox*.  I used Firefox v79 on a Mac.  The page breaks should appear where the `span` was inserted.

The main problem with this approach is that an **additonal blank page** is inserted where the page breaks are  indicated.  I am not expert enough to explain why this happens. This is clearly a problem when printing to paper but the added pages can easily be removed by hand from a PDF. 

The reason why this works is that divs with `display: flex` directives override `page-break` directives, as mentioned [here](https://stackoverflow.com/questions/4884380/css-page-break-not-working-in-all-browsers). Testing revealed that only `div.inner_cell` needed to by modified to use `display: block`, and only when printing. Also mentioned in that page is that Firefox deals with "Twitter Bootstrap classes" better than other browsers. This may be why Firefox prints correctly where other browsers don't. Again my lack of expertise fails me here. This approach works with Firefox but not with Safari or Edge.

**Comment by DrDanRyan** [comment]
After latest update, now I need to select both `div.jp-Cell-inputeWrapper` and `div.jp-InputArea` and make them `display: block;` to have it work.

**Comment by ppKrauss** [comment]
see https://print-css.rocks/

**Comment by DGMacA** [comment]
I've been using 
<div style="page-break-after: always;"></div> 
in a Jupyter Notebook 'Raw' cell very successfully -- until a system update a few days ago.
It now doesn't work.  I re-activated this in the past, but have forgotten how to make it work again.
Help, please!

**Comment by DGMacA** [comment]
Thank you – very helpful.
